baybe/utils/augmentation.py

```diff
@@ -0,0 +1,220 @@
+"""Utilities related to data augmentation."""
+
+from collections.abc import Sequence
+from itertools import permutations, product
+
+import pandas as pd
+
+
+def _row_in_df(row: pd.Series | pd.DataFrame, df: pd.DataFrame) -> bool:
+    """Check whether a row is fully contained in a dataframe.
+
+    Args:
+        row: The row to be checked.
+        df: The dataframe to be checked.
+
+    Returns:
+        Boolean result.
+
+    Raises:
+        ValueError: If ``row`` is a dataframe that contains more than one row.
+    """
+    if isinstance(row, pd.DataFrame):
+        if len(row) != 1:
+            raise ValueError(
+                f"{_row_in_df.__name__} can only be called with pd.Series or "
+                f"pd.DataFrames that have exactly one row."
+            )
+        row = row.iloc[0]
+
+    row = row.reindex(df.columns)
+    return (df == row).all(axis=1).any()
+
+
+def df_apply_permutation_augmentation(
+    df: pd.DataFrame,
+    columns: Sequence[Sequence[str]],
```

Comment:
I'd prefer a slighlty different name here since the name `columns` could be confused with the actual columns of `df`

Reply:
Yes, that would be clearer to me :)

Reply:
I can live with `columns`, but if @AVHopp prefers something more descriptive, then I suggest to somehow encode the fact that the columns are coupled into the name, instead of just repeating the type like `columns_sequences`, which carries no semantic meaning. How about something like `coupled_columns` or `column_coupling` or `column_groups` or similar?

Reply:
Perhaps `column_groups` is a good compromise, because it reflects both the fact that we are talking about collections of columns + it at least partly conveys the message that each group sort of undergoes the same operation?

Reply:
renamed to `column_groups`